test/integration/controllermanager/shootstate/shootstate_test.go

```diff
@@ -0,0 +1,239 @@
+				Labels:       map[string]string{testID: testRunID},
+			},
+			Spec: gardencorev1beta1.ShootSpec{
+				SecretBindingName: ptr.To("secretbinding"),
+				CloudProfileName:  ptr.To("cloudprofile1"),
+				Region:            "europe-central-1",
+				Provider: gardencorev1beta1.Provider{
+					Type: "foo-provider",
+					Workers: []gardencorev1beta1.Worker{
+						{
+							Name:    "cpu-worker",
+							Minimum: 3,
+							Maximum: 3,
+							Machine: gardencorev1beta1.Machine{
+								Type: "large",
+							},
+						},
+					},
+				},
+				DNS: &gardencorev1beta1.DNS{
+					Domain: ptr.To("some-domain.example.com"),
+				},
+				Kubernetes: gardencorev1beta1.Kubernetes{
+					Version: "1.32.0",
+				},
+				Networking: &gardencorev1beta1.Networking{
+					Type: ptr.To("foo-networking"),
+				},
+				SeedName: ptr.To("source-seed"),
+			},
+		}
+
+		addFinalizer = func(shootState *gardencorev1beta1.ShootState) {
+			By("Add ShootState finalizer")
+			Expect(controllerutils.AddFinalizers(ctx, testClient, shootState, shootstate.FinalizerName)).To(Succeed())
+		}
+
+		By("Create Shoot")
+		Expect(testClient.Create(ctx, shoot)).To(Succeed())
+
+		patch := client.MergeFrom(shoot.DeepCopy())
+		shoot.Status.SeedName = shoot.Spec.SeedName
+		Expect(testClient.Status().Patch(ctx, shoot, patch)).To(Succeed())
+
+		By("Create ShootState")
+		shootState = &gardencorev1beta1.ShootState{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: testNamespace.Name,
+				Name:      shoot.Name,
+				Labels:    map[string]string{testID: testRunID},
+			},
+		}
+		Expect(testClient.Create(ctx, shootState)).To(Succeed())
+
+		DeferCleanup(func() {
+			By("Delete Shoot")
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, shoot))).To(Succeed())
+
+			By("Delete ShootState")
+			Expect(controllerutils.RemoveFinalizers(ctx, testClient, shootState, shootstate.FinalizerName)).To(Succeed())
+			Expect(client.IgnoreNotFound(testClient.Delete(ctx, shootState))).To(Succeed())
+
+			By("Ensure ShootState is gone")
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(shootState), shootState)
+			}).Should(BeNotFoundError())
+
+			By("Ensure Shoot is gone")
+			Eventually(func() error {
+				return testClient.Get(ctx, client.ObjectKeyFromObject(shoot), shoot)
+			}).Should(BeNotFoundError())
+
+		})
+	})
+
+	When("migrating Shoot to another Seed", func() {
+		var (
+			patch client.Patch
+		)
+
+		BeforeEach(func() {
+			By("Mark Shoot for migration")
+			patch = client.MergeFrom(shoot.DeepCopy())
+			shoot.Spec.SeedName = &targetSeedName
+			Expect(testClient.SubResource("binding").Patch(ctx, shoot, patch)).To(Succeed())
+		})
+
+		When("Shoot last operation is Migrate", func() {
+			BeforeEach(func() {
+				shoot.Status.LastOperation = &gardencorev1beta1.LastOperation{
+					Type:     gardencorev1beta1.LastOperationTypeMigrate,
+					State:    gardencorev1beta1.LastOperationStateProcessing,
+					Progress: 0,
```

Comment:
Remove this

Reply:
Thanks for pointing this redundant assignment, removed it.